[ replaced with a newer version] Unify dynamic shapes APIs naming 2 (expect_true and check)

laithsakka opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #155973
* __->__ #155776
* #154774

The functions guard_lt, guard_equals, and guard_leq work similarly to torch.check and expect_true, but they operate on SymPy expressions. Notably, guard_equals applies local replacements before comparison, which might be better extracted into a separate function.

This pull request standardizes naming conventions to match symbolic_shapes.py. Specifically,
-  it introduces size_vars.expect_true and size_vars.check.
- guard_lt becomes check_lt
- guard_leq becomes check_leq
- guard_equals becomes check_equals

I am also seeing a couple of wrong usages !! that i will fix  in the next PR 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

laithsakka committed:
unify symolic shapes API2

[ghstack-poisoned]

laithsakka commented on torch/_inductor/codegen/cpp.py:
```diff
@@ -3735,7 +3735,7 @@ def _is_valid_indices(
                             call_ranges[tiling_indice], fallback=0
                         )
                         if call_range < factor_lowp:
```
what is the value of calling 
  V.graph.sizevars.check_lt(call_range, factor_lowp)  # type: ignore[arg-type]
  here ?

laithsakka commented on torch/_inductor/codegen/cpp.py:
```diff
@@ -3735,7 +3735,7 @@ def _is_valid_indices(
                         )
                         if call_range < factor_lowp:
-                            V.graph.sizevars.guard_lt(call_range, factor_lowp)  # type: ignore[arg-type]
+                            V.graph.sizevars.check_lt(call_range, factor_lowp)  # type: ignore[arg-type]
```
call_range < factor_lowp could be a sympy expressions?
which also could be false. this code here sounds wrong tbh?? 

Cc @eellison

laithsakka commented on torch/_inductor/codegen/halide.py:
```diff
@@ -648,7 +648,7 @@ def eq(left, right):
         return False
     if a == b:
-        V.graph.sizevars.guard_equals(left, right)
+        V.graph.sizevars.check_equals(left, right)
```
this seems wrong also !

bobrenjc93 replied:
cc @jansel who probably wrote this since it's the halide backend

jansel replied:
Does this still add the guard?  I think it is required for correctness given how this function is written.

laithsakka replied:
this does add a guard./runtime assert. 
i think maybe you wanted
 if hint(a) ==hint(b):
    V.graph.sizevars.check_equals(left, right)
    
or if   V.graph.sizevars.guard_or_false(Eq(a, b))

i will follow up with a fix for those.

laithsakka replied:
oh actually i just saw this one a, b are hints this is fine.
we can change to V.graph.sizevars.guard_or_false(a, b) with out hint access for simplicity my bad @jansel

laithsakka commented on torch/_inductor/codegen/halide.py:
```diff
@@ -664,7 +664,7 @@ def lt(left, right):
         return False
     if a < b:
-        V.graph.sizevars.guard_lt(left, right)
+        V.graph.sizevars.check_lt(left, right)
```
this seems wrong also !!!

laithsakka commented on torch/_inductor/ir.py:
```diff
@@ -3090,14 +3090,14 @@ def _dynamic_reshape_indexer(old_size, new_size, dense_dim: Optional[int] = None
             elif size_hint(size_new) == size_hint(size_old):
                 view_expr.append(var)
-                V.graph.sizevars.guard_equals(size_new, size_old)
+                V.graph.sizevars.check_equals(size_new, size_old)
```
why not just saying elif  guard_or_false(...)??

drisspg commented on torch/_inductor/sizevars.py:
```diff
@@ -379,59 +379,38 @@ def statically_known_power_of_2(self, expr: Expr) -> bool:
+    # The expect/check functions require you to ALREADY KNOW that a particular
+    # condition holds. They are similar to expect_true in symbolic_shapes.py and
+    # torch.check but operates on sympy expressions instead of symnodes.
+    def expect_true(self, expr: Expr) -> bool:
```
can you move this comment into the """ doc string """

laithsakka replied:
done

drisspg commented on torch/_inductor/sizevars.py:
```diff
@@ -379,59 +379,38 @@ def statically_known_power_of_2(self, expr: Expr) -> bool:
-                self.guard_leq(last_var, var)
-            last_var = var
-        return order
+    def check_leq(self, left: Expr, right: Expr) -> None:
```
ditto, when/ how to use?

laithsakka replied:
not worth to add document for this it just call torch.check and I added documentation there.

bobrenjc93 reviewed (approved):
please address driss's comments before landing

laithsakka committed:
Update on "Unify dynamic shapes APIs naming 2 (expect_true and check)"


The functions guard_lt, guard_equals, and guard_leq work similarly to torch.check and expect_true, but they operate on SymPy expressions. Notably, guard_equals applies local replacements before comparison, which might be better extracted into a separate function.

This pull request standardizes naming conventions to match symbolic_shapes.py. Specifically,
-  it introduces size_vars.expect_true and size_vars.check.
- guard_lt becomes check_lt
- guard_leq becomes check_leq
- guard_equals becomes check_equals

I am also seeing a couple of wrong usages !! that i will fix  in the next PR 

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge -i
